log/logger.go

```diff
@@ -57,6 +52,18 @@ type Logger interface {
 	Impl() any
 }
 
+// WithJSONMarshal configures zerolog global json encoding.
+func WithJSONMarshal(marshaler func(v any) ([]byte, error)) {
+	zerolog.InterfaceMarshalFunc = func(i any) ([]byte, error) {
+		switch v := i.(type) {
+		case fmt.Stringer:
```

Comment:
Same here, what if a struct implements stringer and json.Marshaller, we should use that one first.